dirty_equals/_other.py

```diff
@@ -369,3 +373,166 @@ def equals(self, other: Any) -> bool:
+    class Foo:
+        a: int
+        b: int
+        c: str
+
+    foo = Foo(1, 2, 'c')
+
+    assert foo == IsDataclass
+    assert foo == IsDataclass(a=IsInt, b=2, c='c')
+    assert foo == IsDataclass(b=2, a=1).settings(partial=True)
+    assert foo != IsDataclass(a=IsInt, b=2).settings(strict=True)
+    assert foo == IsDataclass(a=IsInt, b=2).settings(strict=True, partial=True)
+    assert foo != IsDataclass(b=2, a=1).settings(strict=True, partial=True)
+    ```
+    """
+
+    def __init__(self, **fields: Any):
+        """
+        Args:
+            fields: key-value pairs of field-value to check for.
+        """
+        self.strict = False
+        self.partial = False
+        self._post_init()
+        super().__init__(**fields)
+
+    def _post_init(self) -> None:
+        pass
+
+    def equals(self, other: Any) -> bool:
+        if is_dataclass(other) and not isinstance(other, type):
+            if self._repr_kwargs:
+                return self._fields_check(other)
+            else:
+                return True
+        else:
+            return False
+
+    def settings(
+        self,
+        *,
+        strict: bool | None = None,
+        partial: bool | None = None,
+    ) -> IsDataclass:
+        """Allows to customise the behaviour of `IsDataclass`, technically a new `IsDataclass` to allow chaining."""
+        new_cls = self.__class__(**self._repr_kwargs)
+        new_cls.__dict__ = self.__dict__.copy()
+
+        if strict is not None:
+            new_cls.strict = strict
+        if partial is not None:
+            new_cls.partial = partial
+
+        return new_cls
+
+    def _fields_check(self, other: Any) -> bool:
+        """
+        Checks exactness of fields using [`IsDict`][dirty_equals.IsDict] with given settings.
+
+        Remark that if this method is called, then `other` is an instance of a dataclass, therefore we can call
+        `dataclasses.asdict` to convert to a dict.
+        """
+        return asdict(other) == IsDict(self._repr_kwargs).settings(strict=self.strict, partial=self.partial)
+
+
+class IsPartialDataclass(IsDataclass):
+    """
+    Inherits from [`IsDataclass`][dirty_equals.IsDataclass] with `partial=True` by default.
+
+    ```py title="IsPartialDataclass"
+    from dataclasses import dataclass
+    from dirty_equals import IsInt, IsPartialDataclass
+
+    @dataclass
+    class Foo:
+        a: int
+        b: int
+        c: str = 'c'
+
+    foo = Foo(1, 2, 'c')
+
+    assert foo == IsPartialDataclass
+    assert foo == IsPartialDataclass(a=1)
+    assert foo == IsPartialDataclass(b=2, a=IsInt)
+    assert foo != IsPartialDataclass(b=2, a=IsInt).settings(strict=True)
+    assert Foo != IsPartialDataclass
+    ```
+    """
+
+    def _post_init(self) -> None:
+        self.partial = True
+
+
+class IsStrictDataclass(IsDataclass):
+    """
+    Inherits from [`IsDataclass`][dirty_equals.IsDataclass] with `strict=True` by default.
+
+    ```py title="IsStrictDataclass"
```

Comment:
I am not sure about what it's missing here. 
Error line is blank in report:

| File | Function | Function Line | Error Line | Error |
| ----| ----------| --------------- | ---------- | ----- |  
| tests/test_docs.py  |  test_docstrings[dirty_equals/_other.py:518-534]  |  7              |              |   |

Reply:
most likely formatting, you can probably correct it by running that test with `--update-examples`.

The report line might be blank, but there's details of the failure just above.